tests/53-sim-binary_tree.c

```diff
@@ -49,38 +93,26 @@ int main(int argc, char *argv[])
 		goto out;
 	}
 
-	rc = seccomp_arch_remove(ctx, SCMP_ARCH_NATIVE);
-	if (rc < 0)
-		goto out;
-	rc = seccomp_arch_add(ctx, SCMP_ARCH_X86_64);
-	if (rc < 0)
-		goto out;
-	rc = seccomp_arch_add(ctx, SCMP_ARCH_X86);
-	if (rc < 0)
-		goto out;
 	rc = seccomp_attr_set(ctx, SCMP_FLTATR_CTL_OPTIMIZE, 2);
 	if (rc < 0)
 		goto out;
 
-	/* NOTE: this test is entirely fabricated and should not be
-	 * 	 replicated in the real world.
-	 *
-	 *	 The MAX_SYSCALL number (330) was chosen to force seccomp to
-	 *	 build an unbalanced binary tree - and it happens to be less
-	 *	 than the current syscall max.  The syscall numbers are
-	 *	 hardcoded to simplify the test.  A few syscalls have
-	 *	 argument chains to further complicate the filter.
-	 */
-
-	for (i = 0; i < MAX_SYSCALL; i++) {
+	for (i = 0; i < table_size; i++) {
 		/* arbitrarily make the filter more complex by filtering
 		 * on arguments for a few syscalls
 		 */
-		if (i == 10 || i == 53 || i == 61 || i == 255)
-			rc = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(i), i, 1,
-					      SCMP_A0(SCMP_CMP_EQ, i));
+		if (table[i].syscall == SCMP_SYS(mprotect) ||
+		    table[i].syscall == SCMP_SYS(dup) ||
+		    table[i].syscall == SCMP_SYS(alarm))
```

Comment:
Why not add an additional field to the syscall table to indicate that this syscall should have an argument added to the filter?